### DISCUSSION TITLE
State/City name normalization

### DISCUSSION THREAD

**Discussion by joaorb64** [dr5hn/countries-states-cities-database#606] [Ideas]
Hi,

I was wondering if there's a standard for city name language.

If I check the states in Brazil, the Federal District is identified by its name in Portuguese, Distrito Federal.

Now, if I check Portugal, the capital Lisbon is identified by its English name Lisbon instead of the Portuguese Lisboa.

I was wondering what would be the better way of settling a standard: adding localization to state/city names or to have everything in its own language. Or having everything in English? Opening this issue for discussion.

**Comment by dr5hn** [comment]
Hi,

Thank you for raising this issue and initiating the discussion. The question of standardizing city name language is indeed important.

In my perspective, I believe it would be beneficial to establish a standard where the main name of a city is kept in English. This ensures consistency across different locations and allows for easier recognition and understanding for a global audience.

However, it could also be valuable to create a separate field to store the native name of a city. This way, we can accommodate the local language and provide a comprehensive representation of the city's identity.

By implementing this approach, we can strike a balance between maintaining a standardized naming convention and respecting the cultural and linguistic diversity of different regions.

I welcome further input and discussion from others to reach a consensus on this matter.

Thank you.

**Reply by relense** [reply]
Hi, I came accross this issue because the capital of Portugal is in english. 

> However, it could also be valuable to create a separate field to store the native name of a city. This way, we can accommodate the local language and provide a comprehensive representation of the city's identity.

I agree that this should have some kind of native field associated. I understand that in the case of Lisbon it makes sense to have the name that most people in the world associate with, but if I want to reach the Portuguese audience in Portugal and use this database, they won't find the name of their city because they would search in their native language. 

This can also happen for everyone that searchs in their native language in any country.
If someone builds an app and uses this, a native person might not know english and won't find what they want in their native tongue. This becomes an issue.

**Comment by pfrenssen** [comment]
I have some thoughts about this, and I think having just the English name + the native name is not sufficient.

- There are cities in multilingual regions which may have multiple "native" names.
    Example: "Brussels" - since Belgium has 3 official languages, in native language this is either "Brussel" (Dutch), "Bruxelles" (French) or "Brüssel" (German).

- There are also cities which have dedicated names in many different languages, not just in English and their native language(s). Especially bigger cities / tourist destinations / cities close to a border.
    Example: London - known internationally as Lontoo, Londain, Londres, Londra, Лондон, Londonas, 倫敦, ロンドン, ...

So maybe instead of just English + native, we can maintain a list of translations keyed by langcode?

```
  -
    id: 50388
    name: London
    state_id: 2336
    state_code: ENG
    state_name: England
    country_id: 232
    country_code: GB
    country_name: 'United Kingdom'
    latitude: '51.50853000'
    longitude: '-0.12574000'
    wikiDataId: Q84
    translations:
      kr: 런던
      pt: Londres
      nl: Londen
      de: London
      es: Londres
```
